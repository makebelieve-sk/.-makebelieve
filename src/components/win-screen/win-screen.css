.result {
    display: flex;
    align-items: center;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
    height: 100%; 
}
.result__logo {
    margin: 0;
    margin-top: 20px; 
    margin-bottom: 20px;
}
.result__title {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    font-size: 33px;
    color: #230d1a;
    text-align: center;
}
.result__total {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    color: #230d1a;
    text-align: center;
    font-size: 28px;
    width: 500px; 
}
.result__total--fail {
    margin-top: 170px; 
}
.result__text {
    margin: 0;
    margin-top: 10px;
    font-size: 16px; 
}
.result__replay {
    position: relative;
    padding: 0;
    color: #ff9749;
    font-size: 28px;
    border: none; 
    background: transparent; 
}
.result__replay::after {
    position: absolute;
    content: "";
    width: 27px;
    height: 25px;
    top: 50%;
    right: 0;
    background-position: -5px -357px;
    -webkit-transform: translateY(-50%);
            transform: translateY(-50%); 
}
.result__replay:hover, .result__replay:focus {
    color: #f00000;
    cursor: pointer; 
}
.result__replay:hover::after, .result__replay:focus::after {
    background-position: -5px -392px; 
}