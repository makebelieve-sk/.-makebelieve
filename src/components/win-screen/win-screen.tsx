import React from 'react';
import { Link } from 'react-router-dom';

import './win-screen.css';
import '../authorization-screen/authorization-screen.css';

type WinScreenType = {
    isAuthorization: boolean, 
    mistakes: number,
    step: number,
    minutes: string | number,
    seconds: string | number,
    reset: () => any
};

export const WinScreen: React.FC<WinScreenType> = ({ isAuthorization, mistakes, step, minutes, seconds, reset }) => {
    let resMinutes = Number(minutes);
    if (seconds < 10) {
        seconds = Number(seconds);
    }

    return (
        <div id="result-success" className="result-success">
            <section className="result">
                <div className="result__logo"><img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" /></div>
                <h2 className="result__title">Вы настоящий меломан!</h2>
                <p className="result__total">За {resMinutes} минуты и {seconds} секунд вы набрали { step } балла, совершив {mistakes} ошибки</p>
                { isAuthorization ? 
                    <p className="result__text">Вы заняли 2 место из 10. Это лучше чем у 80% игроков</p> :
                    <> 
                        <p className="result__text">Хотите сравнить свой результат с предыдущими попытками?</p>
                        <p className="result__text">Представтесь!</p>
                        <form className="login_form" action="">
                            <p className="login_field">
                                <label className="login_label" htmlFor="name">Логин</label>
                                <input className="login_input" type="text" name="name" id="name" />
                            </p>
                            <p className="login_field">
                                <label className="login_label" htmlFor="password">Пароль</label>
                                <input className="login_input" type="text" name="password" id="password" />
                                <span className="login_error">Неверный пароль</span>
                            </p>
                            <Link to="/" onClick={reset} className="login_button button" type="submit">Войти</Link>
                        </form>
                    </>
                } 
                <Link to="/" className="result__replay" type="button" onClick={reset}>Сыграть ещё раз</Link>
            </section>
        </div>
    );
};