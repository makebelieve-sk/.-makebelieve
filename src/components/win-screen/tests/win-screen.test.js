import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Router } from 'react-router-dom';

import { WinScreen } from '../win-screen';
import history from '../../../history';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2, prop3, prop4, prop5, prop6) => shallow(
    <WinScreen isAuthorization={prop1} mistakes={prop2} step={prop3} minutes={prop4} seconds={prop5} reset={prop6} />
);

const mock = {
    isAuthorization: false, 
    mistakes: 0,
    step: 0,
    minutes: '01',
    seconds: '01',
    reset: jest.fn()
};

describe(`WinScreen component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.isAuthorization, mock.mistakes, mock.step, mock.timeEnd, mock.seconds, mock.reset);
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.result-success');
        expect(wrapper.length).toBe(1);
    });

    it(`WinScreen correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(
                <Router history={history}><WinScreen 
                    isAuthorization={mock.isAuthorization}
                    mistakes={mock.mistakes}
                    step={mock.step}
                    timeEnd={mock.timeEnd}
                    seconds={mock.seconds}
                    reset={mock.reset}
                />
                </Router>
            )
            .toJSON();
    
        expect(tree).toMatchSnapshot();
    });
});