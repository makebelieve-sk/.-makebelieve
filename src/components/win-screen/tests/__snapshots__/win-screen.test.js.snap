// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinScreen component WinScreen correctly renders after relaunch 1`] = `
<div
  className="result-success"
  id="result-success"
>
  <section
    className="result"
  >
    <div
      className="result__logo"
    >
      <img
        alt="Угадай мелодию"
        height="83"
        src="img/melody-logo.png"
        width="186"
      />
    </div>
    <h2
      className="result__title"
    >
      Вы настоящий меломан!
    </h2>
    <p
      className="result__total"
    >
      За 
      NaN
       минуты и 
      1
       секунд вы набрали 
      0
       балла, совершив 
      0
       ошибки
    </p>
    <p
      className="result__text"
    >
      Хотите сравнить свой результат с предыдущими попытками?
    </p>
    <p
      className="result__text"
    >
      Представтесь!
    </p>
    <form
      action=""
      className="login_form"
    >
      <p
        className="login_field"
      >
        <label
          className="login_label"
          htmlFor="name"
        >
          Логин
        </label>
        <input
          className="login_input"
          id="name"
          name="name"
          type="text"
        />
      </p>
      <p
        className="login_field"
      >
        <label
          className="login_label"
          htmlFor="password"
        >
          Пароль
        </label>
        <input
          className="login_input"
          id="password"
          name="password"
          type="text"
        />
        <span
          className="login_error"
        >
          Неверный пароль
        </span>
      </p>
      <button
        className="login_button button"
        type="submit"
      >
        Войти
      </button>
    </form>
    <a
      className="result__replay"
      href="/"
      onClick={[Function]}
      type="button"
    >
      Сыграть ещё раз
    </a>
  </section>
</div>
`;
